polytope_server/common/metric_store/dynamodb_metric_store.py

```
@@ -0,0 +1,207 @@
+                raise ValueError("Request already exists in request store") from e
+            raise
+
+    def remove_metric(self, uuid):
+        try:
+            self.table.delete_item(Key={"uuid": str(uuid)}, ConditionExpression=Attr("uuid").exists())
+        except botocore.exceptions.ClientError as e:
+            if e.response["Error"]["Code"] == "ConditionalCheckFailedException":
+                raise KeyError("Request does not exist in request store") from e
+            raise
+
+    def get_metric(self, uuid):
+        response = self.table.get_item(Key={"uuid": str(uuid)})
+        if "Item" in response:
+            return _load(response["Item"])
+
+    def get_metrics(self, ascending=None, descending=None, limit=None, request_id=None, **kwargs):
+        if ascending is not None and descending is not None:
+            raise ValueError("Cannot sort by ascending and descending at the same time.")
+
+        if request_id is not None:
+            fn = self.table.query
+            params = {
+                "IndexName": "request-index",
+                "KeyConditionExpression": Key("request_id").eq(request_id),
+            }
+        else:
+            fn = self.table.scan
+            params = {}
+
+        if limit is not None:
+            params["Limit"] = limit
+
+        if query := _make_query(**kwargs):
+            params["FilterExpression"] = reduce(operator.__and__, (Attr(key).eq(value) for key, value in query.items()))
+        
+        items = (_load(item) for item in _iter_items(fn, **params))
+        if ascending is not None:
+            return sorted(items, key=lambda item: getattr(item, ascending))
+        if descending is not None:
+            return sorted(items, key=lambda item: getattr(item, descending), reverse=True)
+        return list(items)
+
+
+    def update_metric(self, metric):
+        self.table.put_item(Item=_dump(metric))
+
+    def wipe(self):
+        pass
```

Should we raise an error in case someone try to use it?

---

Perhaps a warning